trl/trainer/grpo_trainer.py

```diff
@@ -627,9 +627,16 @@ def data_collator(features):  # No data collation is needed in GRPO
                 )
 
             if self.accelerator.is_main_process:
-                self.vllm_client = VLLMClient(
-                    args.vllm_server_host, args.vllm_server_port, connection_timeout=args.vllm_server_timeout
-                )
+                if args.vllm_server_base_url is not None:
+                    self.vllm_client = VLLMClient(
+                        base_url=args.vllm_server_base_url, connection_timeout=args.vllm_server_timeout
+                    )
+                else:
+                    self.vllm_client = VLLMClient(
+                        host=args.vllm_server_host, 
+                        server_port=args.vllm_server_port, 
+                        connection_timeout=args.vllm_server_timeout
+                    )
```

Comment:
maybe this is easier?

```suggestion
                base_url = args.vllm_server_base_url if args.vllm_server_base_url is not None else f"http://{args.vllm_server_port}:{args.vllm_server_port}"
                self.vllm_client = VLLMClient(base_url=ase_url, connection_timeout=args.vllm_server_timeout)
```